backend/src/main/java/org/cryptomator/hub/api/UsersResource.java

```diff
@@ -62,6 +67,33 @@ public Response putMe(@Nullable @Valid UserDto dto) {
 		return Response.created(URI.create(".")).build();
 	}
 
+	@POST
+	@Path("/me/access-tokens")
+	@RolesAllowed("user")
+	@Transactional
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Operation(summary = "adds/updates user-specific vault keys", description = "Stores one or more vaultid-vaultkey-tuples for the currently logged-in user, as defined in the request body ({vault1: token1, vault2: token2, ...}).")
+	@APIResponse(responseCode = "200", description = "all keys stored")
+	public Response updateMyAccessTokens(@NotEmpty Map<UUID, String> tokens) {
```

Comment:
It is important, that if the user does not have any access tokens, he/she does not get an error returned. Otherwise our client application has to deal with this edge case.

Reply:
> nor do i see the necessity to be non-empty

Indeed it will be a no-op. I'll remove this constraint (copy-pasted anyway)

Reply:
changed in 58ff0d684ba1eda836320430e58b44b48029baff